File: gmstudio/gmextras/converts.go
package gmextras

import (
	"github.com/MaxKlaxxMiner/golangMusicStudio/gmstudio/gmconst"
)

func ConvertIntSamplesToFloat32(dst []float32, src []int32) {
	if len(dst) != len(src) {
		panic("invalid buffer sizes")
	}
	for i := 0; i < len(dst); i++ {
		dst[i] = float32(src[i]) / float32(1<<(gmconst.DynamicBits-1))
	}
}

File: gmstudio/gmextras/converts.c
// wasm-compiler: https://wasdk.github.io/WasmFiddle/
typedef int int32;
typedef unsigned int uint32;
typedef long long int64;
typedef unsigned long long uint64;
#define SampleBits 32
#define DynamicBits 24

void convertIntSamplesToFloat32(float* dst, int32* src, int sampleCount) {
  for (int i = 0; i < sampleCount; i++) {
    dst[i] = (float)src[i] / (float)(1 << (DynamicBits - 1));
  }
}
